mypy failing on main branch

After upgrading torch to 2.6.0, I see the following mypy test failures on `main`

```pytb
$ make typecheck
mypy cellarium tests
cellarium/ml/layers/head.py:48: error: Argument 1 has incompatible type "Tensor | Module"; expected "Tensor"  [arg-type]
cellarium/ml/layers/head.py:51: error: Argument 1 to "zeros_" has incompatible type "Tensor | Module"; expected "Tensor"  [arg-type]
cellarium/ml/layers/embedding.py:46: error: Argument 1 has incompatible type "Tensor | Module"; expected "Tensor"  [arg-type]
cellarium/ml/layers/embedding.py:89: error: Argument 1 has incompatible type "Tensor | Module"; expected "Tensor"  [arg-type]
cellarium/ml/models/onepass_mean_var_std.py:128: error: Unsupported operand types for + ("Tensor" and "None")  [operator]
cellarium/ml/models/onepass_mean_var_std.py:128: note: Right operand is of type "Tensor | None"
```

--- a/cellarium/ml/layers/embedding.py
+++ b/cellarium/ml/layers/embedding.py
@@ -43,6 +43,7 @@ class GeneExpressionEmbedding(nn.Module):
 
     def _reset_parameters(self) -> None:
         for module in self.E.children():
+            assert isinstance(module.weight, torch.Tensor)
             create_initializer(self.embeddings_initializer)(module.weight)
 
     def forward(self, gene_tokens_nc: dict[str, torch.Tensor]) -> torch.Tensor:
@@ -86,6 +87,7 @@ class MetadataEmbedding(nn.Module):
 
     def _reset_parameters(self) -> None:
         for module in self.E.children():
+            assert isinstance(module.weight, torch.Tensor)
             create_initializer(self.embeddings_initializer)(module.weight)
 
     def forward(self, metadata_tokens_n: dict[str, torch.Tensor]) -> torch.Tensor:

--- a/cellarium/ml/layers/head.py
+++ b/cellarium/ml/layers/head.py
@@ -45,9 +45,11 @@ class MultiHeadReadout(nn.Module):
 
     def _reset_parameters(self) -> None:
         for module in self.W.children():
+            assert isinstance(module.weight, torch.Tensor)
             create_initializer(self.heads_initializer)(module.weight)
 
             if module.bias is not None:
+                assert isinstance(module.bias, torch.Tensor)
                 nn.init.zeros_(module.bias)
 
     def forward(self, hidden_state_ncd: torch.Tensor) -> dict[str, torch.Tensor]:

--- a/cellarium/ml/models/onepass_mean_var_std.py
+++ b/cellarium/ml/models/onepass_mean_var_std.py
@@ -125,8 +125,8 @@ class OnePassMeanVarStd(CellariumModel):
         """
         mean_g = self.x_sums / self.x_size
         if self.algorithm == "shifted_data":
-            if self.x_shift is not None:
-                mean_g = mean_g + self.x_shift
+            assert isinstance(self.x_shift, torch.Tensor)
+            mean_g = mean_g + self.x_shift
         return mean_g
 
     @property

--- a/cellarium/ml/models/onepass_mean_var_std.py
+++ b/cellarium/ml/models/onepass_mean_var_std.py
@@ -125,7 +125,8 @@ class OnePassMeanVarStd(CellariumModel):
         """
         mean_g = self.x_sums / self.x_size
         if self.algorithm == "shifted_data":
-            mean_g = mean_g + self.x_shift
+            if self.x_shift is not None:
+                mean_g = mean_g + self.x_shift
         return mean_g
 
     @property

